tests/loading_from_clipboard/loading_from_clipboard_test.cpp

```diff
@@ -0,0 +1,66 @@
+#include "gtest/gtest.h"
+#include "loading_from_clipboard_test.h"
+#include <QTextStream>
+#include "../../common/decklist.h"
+
+DeckList *fromClipboard(QString *clipboard);
+
+DeckList *fromClipboard(QString *clipboard) {
+    DeckList *deckList = new DeckList;
+    QTextStream *stream = new QTextStream(clipboard);
+    deckList->loadFromStream_Plain(*stream);
+    return deckList;
+}
+
+struct AssertDeckList {
+    QList<QString> expectedCards;
+    QList<QString> actualCards;
+
+    AssertDeckList(QList<QString> _expectedCards) : expectedCards(_expectedCards) {}
+
+    void operator()(const InnerDecklistNode *, const DecklistCardNode *card) {
+        for (int i = 0; i < card->getNumber(); i++) {
+            actualCards.append(card->getName());
+        }
+    }
+
+    void makeAssertions() {
+        ASSERT_EQ(expectedCards.size(), actualCards.size());
+        ASSERT_EQ(expectedCards, actualCards);
+    }
+};
+
+namespace {
+    TEST(LoadingFromClipboardTest, EmptyDeck) {
+        DeckList *deckList = fromClipboard(new QString(""));
+
+        QStringList cardsInDeck = deckList->getCardList();
+        ASSERT_TRUE(cardsInDeck.isEmpty()) << "Deck should be empty";
+    }
+
+    TEST(LoadingFromClipboardTest, EmptySideboard) {
+        DeckList *deckList = fromClipboard(new QString("Sideboard"));
+
+        QStringList cardsInDeck = deckList->getCardList();
+        ASSERT_TRUE(cardsInDeck.isEmpty()) << "Deck should be empty";
+    }
+
+    TEST(LoadingFromClipboardTest, QuantityPrefixed) {
+        QString * clipboard = new QString("");
+        clipboard->append("1 Mountain\n");
+        clipboard->append("2x Island\n");
+        DeckList *deckList = fromClipboard(clipboard);
+
+        QList<QString> expectedDeck = QList<QString>() << "Mountain" << "Island" << "Island";
+        AssertDeckList assertDeckList(expectedDeck);
+
+        deckList->forEachCard(assertDeckList);
```

Comment:
I'd like to avoid nesting too deeply - what about one QMap<QString, int> for the mainboard and one QMap<QString, int> for the sideboard? AssertDeckList can have two constructors, one that only takes a mainboard and one that takes a mainboard and sideboard.